### DISCUSSION TITLE
How to check the test accuracy of a model for each of hyperparameter setting in AutoGluon?

### DISCUSSION THREAD

**Discussion by Zay-Ben** [autogluon/autogluon#4295] [Q&A]
Dear Fellow Data Science Enthusiasts,

This may sound a silly question, but I would appreciate any response on it. May I ask how to check the test accuracy of a model for each of hyperparameter setting in AutoGluon? In Non-AutoML pipeline, we get to train a model over a set of hyperparameter settings. For each setting, we get to evaluate the accuracy of the model over a test set (not cross validation set) and a train set. Then chose which hypermeter setting that is neither underfitting nor overfitting. Thank you!

Benz

**Comment by shchur** [comment]
Hi @Zay-Ben, the [`TabularPredictor.leaderboard`](https://auto.gluon.ai/stable/api/autogluon.tabular.TabularPredictor.leaderboard.html) method allows evaluating all models on a new test set. You can see the config of each model by setting `predictor.leaderboard(test_data, extra_info=True)`
